pages/index.js

```diff
@@ -57,14 +59,27 @@ export default function Home() {
     },
   ]);
 
+  const [mapShown, setMapShown] = useState(true);
+  console.log(mapShown);
+
+  function handleMapShown() {
+    setMapShown(!mapShown);
+  }
+
   return (
     <>
-      <Map></Map>
-      <ListSurfspots>
-        {surfspots.map((surfspot) => {
-          return <ListSurfspotsItems name={surfspot.name} key={surfspot.ID} />;
-        })}
-      </ListSurfspots>
+      <Header mapShown={handleMapShown} />
```

Comment:
Maybe consider renaming the function prop - "onMapShown" or something. mapShown is already the name of your state variable. It doesn't conflict here, but the code is better readable when different variables don't match in name